amethyst_assets/src/loader_new.rs

```diff
@@ -60,22 +65,30 @@ pub struct GenericHandle {
     chan: Arc<Sender<RefOp>>,
     id: LoadHandle,
 }
+
 impl GenericHandle {
     fn new(chan: Arc<Sender<RefOp>>, handle: LoadHandle) -> Self {
         Self { chan, id: handle }
     }
 }
+
 impl Drop for GenericHandle {
     fn drop(&mut self) {
         self.chan.send(RefOp::Decrease(self.id))
     }
 }
+
 impl AssetHandle for GenericHandle {
     fn load_handle(&self) -> &LoadHandle {
         &self.id
     }
 }
 
+/// Handle to an asset that does not prevent the asset from being unloaded.
+///
+/// This is useful for balancing performance and memory usage -- frequently or recently used assets
+/// may be held with strong references, and assets that are not used frequently or recently are
+/// downgraded to weak handles so that they may be freed when memory usage is reaching a set limit.
```

Comment:
This is not true - anything that reaches 0 in reference count is guaranteed to be immediately unloaded, so the usage of the asset has no relation to when it will be unloaded. Weak handles are thus primarily used when you kind of *want to use something that is already loaded*. I.e. a strong handle to an asset may be guaranteed to exist elsewhere in the program, maybe the loading screen, and so you can simply get and use a weak handle to that asset in other parts of your code. This removes reference counting overhead, for example, but also ensures that the system which uses the weak handle is not in control of when to unload the asset.

Reply:
ah of course, that last part is also important